src/base/Type.ml

```diff
@@ -208,6 +227,25 @@ module MkType (I : ScillaIdentifier) = struct
   (*******************************************************)
   module IdLoc_Comp = IdLoc_Comp (TIdentifier)
 
+  (* The types of addresses we care about.
+   * Lattice:
+        AnyAddr
+           |
+        CodeAddr
+          / \
+    LibAddr CodeAddr
+   *)
+  type 'a addr_kind =
+    (* Any address in use. *)
+    | AnyAddr
+    (* Address containing a library or contract. *)
+    | CodeAddr
+    (* Address containing a library. *)
+    | LibAddr
+    (* Address containing a contract. *)
+    | ContrAddr of 'a IdLoc_Comp.Map.t
+  [@@deriving sexp]
```

Comment:
Looks like this definition is duplicated here. Is it possible to move it to its own (parameterized) module?